pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/security/ApexCRUDViolationRule.java

```diff
@@ -636,12 +657,18 @@ private boolean validateCRUDCheckPresent(final ApexNode<?> node, final Object da
         boolean isImproperDMLCheck = !isProperESAPICheckForDML(typeCheck, crudMethod)
                 && !isProperAuthPatternBasedCheckForDML(typeCheck, crudMethod);
         boolean noSecurityEnforced = !isWithSecurityEnforced(node);
+        boolean userMode = isWithUserMode(node);
+        boolean systemMode = isWithSystemMode(node);
         if (missingKey) {
             //if condition returns true, add violation, otherwise return.
             if (isImproperDMLCheck && noSecurityEnforced) {
                 addViolation(data, node);
                 return true;
             }
+            if (isImproperDMLCheck && !userMode && !systemMode) {
```

Comment:
We probably need to add `!noSecurityEnforced` to the condition, so that we don't issue a violation, when someone is using "WITH SECURITY ENFORCED". That's probably the reason for the test failures.

I'm going to fix this myself.